src/poetry/console/commands/add.py

```diff
@@ -230,6 +231,15 @@ def handle(self) -> int:
                 )
             )
 
+        if self.poetry.config.get("dependencies.sort") is True:
+            sorted_dependencies = OrderedDict(sorted(section.items()))
```

Comment:
One more issue here: casting `section` to dictionary causes loss of comments in project file. That might be critical for some projects. Please, try to fix the issue

Reply:
> One more issue here: casting `section` to dictionary causes loss of comments in project file. That might be critical for some projects. Please, try to fix the issue

If comments were preserved, where would they go? We could probably preserve inline comments, but single line comments would be more difficult.

The most common use case for comments that I have seen is breaking up dependencies into subsections, like this:

```toml
[tool.poetry.dev-dependencies]
# tests
pytest = "*"
# docs
mkdocs = "*"
```

If the dependencies were sorted, where would the comments go? It would probably end up like this:

```toml
[tool.poetry.dev-dependencies]
# tests
# docs
mkdocs = "*"
pytest = "*"
```

A more effective approach might be to remove the comments and separate these commented sections into [Poetry 1.2 dependency groups](https://python-poetry.org/docs/managing-dependencies/#dependency-groups).

```toml
[tool.poetry.group.test.dependencies]
pytest = "*"

[tool.poetry.group.docs.dependencies]
mkdocs = "*"
```

I'm happy to try preserving comments if there is a suggested data structure from TOML Kit.

Reply:
Also comments might be used to temporarily disable a dependency or explain a dependency directly below the comment.
In these cases, it's quite important to keep those comments. I think it would be good to make the comments stick to the next dependency line:

```
[tool.poetry.dependencies]
simplejson = "^3.17.5"  # used in requests to serialize decimals
# Consider to replace requests with httpx 
# to improve performance.
# httpx = "^0.23.0"
requests = "^2.26.0"
fastapi = "^1"
```

should turn into

```
[tool.poetry.dependencies]
fastapi = "^1"
# Consider to replace requests with httpx 
# to improve performance.
# httpx = "^0.23.0"
requests = "^2.26.0"
simplejson = "^3.17.5"  # used in requests to serialize decimals
```

Your example is about a case that can be resolved using poetry functional, as you mentioned. Users should be responsible for proper use of Poetry. So I would keep comments in this case as well

Reply:
I don't know how to use tomlkit properly. But I can just share my thoughts.

It seems comment lines are stored in [`section.value.body`](https://github.com/sdispater/tomlkit/blob/master/tomlkit/container.py#L46) list with `None` as a key and `Comment(..)` as a value. You can't get an access to the comment lines via `section.items()` method. I didn't find any other way to access to the comments but `section.value.body`. And I don't see methods to remove comment lines from the section (neither `section.remove()` nor `section.clear()` don't remove the comments). So, you should re-create an entire section and append sorted lines into it, or try to sort `section.value.body` in-place.